figma-shop/src/app/shared/services/auth.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import { AngularFireAuth } from '@angular/fire/auth';
+import { AngularFirestore, AngularFirestoreDocument } from '@angular/fire/firestore';
+import * as firebase from 'firebase/app';
+import { Observable, of } from 'rxjs';
+import { UserInterface } from '../interfaces/user.interface';
+import { switchMap } from 'rxjs/operators';
+import { SnackbarService } from './snackbar.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthService {
+  private currentUser: any;
+  public user: Observable<UserInterface>;
+
+  constructor(
+    private afAuth: AngularFireAuth,
+    private afFirestore: AngularFirestore,
+    private snackbarService: SnackbarService
+  ) {
+    this.user = this.afAuth.authState.pipe(
+      switchMap(user => {
+        if (user) {
+          return this.afFirestore.doc(`userList/${user.uid}`).valueChanges();
+        } else {
+          return of(null);
+        }
+      })
+    );
```

Comment:
Please write it in a private method and call it wherever you want

Reply:
got it